syntax = "proto3";

package polygrasp;

service GraspServer {
  rpc GetGrasps(stream PointCloud) returns (stream GraspGroup) {}
}

service PointCloudServer {
  rpc GetPointcloud(stream Image) returns (stream PointCloud) {}
  rpc SegmentPointcloud(stream PointCloud) returns (stream ObjectMask) {}
}

message Image {
  bytes rgbd = 1;
}

message ObjectMask {
  int64 id = 1;
  bytes mask = 2;
}

message PointCloud {
  bytes pcd = 1;
  bytes color = 2;
}
  
message GraspGroup {
  bytes grasp_group_array = 1;
}
